### PR TITLE
Advance routines

### PR DESCRIPTION
**Pull request by Thuener**
Add routines of Gurobi to get information from tableau

### CODE REVIEW

File: src/grb_advroutines.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
+mutable struct GRBsvec
+    len::Cint
+    ind::Ptr{Cint}
+    val::Ptr{Cdouble}
+end
+
+function get_tableaurow_grb(model::Model, cidx::Integer)
+	len = num_vars(model)+num_constrs(model)
+	idx = Array{Cint}(len)
+	val = Array{Cdouble}(len)
```
Comment (odow) [inline src/grb_advroutines.jl]: Looks like the indentation is off?

Comment (Thuener) [reply src/grb_advroutines.jl]: What is the default here? Spaces or tabs?

Comment (joehuchette) [reply src/grb_advroutines.jl]: 4 spaces

File: src/grb_advroutines.jl [hunk 2]
```diff
@@ -0,0 +1,53 @@
+mutable struct GRBsvec
+    len::Cint
+    ind::Ptr{Cint}
+    val::Ptr{Cdouble}
+end
+
+function get_tableaurow_grb(model::Model, cidx::Integer)
```
Comment (mlubin) [inline src/grb_advroutines.jl]: It's best for this function to have a name that matches the function being wrapped. See e.g., https://github.com/JuliaOpt/Gurobi.jl/blob/93b4d3e37c3d8ed2ec37b59d207504537e85e173/src/grb_quad.jl#L4.

Comment (Thuener) [reply src/grb_advroutines.jl]: BinvRowi is a strange name for a function in Julia. The other problem is that this name is misleading, this is not actually the row from B^-1 is the row from A B^-1. http://www.gurobi.com/documentation/7.5/refman/c_grbbinvrowi.html
Should I call it ABinv_row?

Comment (odow) [reply src/grb_advroutines.jl]: How about `c_api_binvrowi`

File: src/grb_advroutines.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        throw(GurobiError(model.env, ret))
+    end
+    return idx[1:grb_v.len]+1, val[1:grb_v.len]
+end
+
+function get_tableaurow!(row::Vector{Float64}, model::Model, cidx::Integer)
+    idx, val = get_tableaurow_grb(model, cidx)
+    row[idx] = val
+    return nothing
+end
```
Comment (odow) [inline src/grb_advroutines.jl]: I'd prefer not to have these helper functions in Gurobi.jl. They don't get tested and lead to multiple ways to do the same thing.

File: src/grb_advroutines.jl [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+function get_tableaurow!(BinvA::Matrix{Float64}, model::Model, cidx::Integer)
+    idx, val = get_tableaurow_grb(model, cidx)
+    BinvA[cidx, idx] = val
+    return nothing
+end
```
Comment (odow) [inline src/grb_advroutines.jl]: Ditto

File: src/grb_advroutines.jl [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+function get_tableaurow(model::Model, cidx::Integer)
+    len = num_vars(model)+num_constrs(model)
+    spv = zeros(len)
+    get_tableaurow!(spv, model, cidx)
+    return spv
+end
```
Comment (odow) [inline src/grb_advroutines.jl]: Ditto

File: src/grb_advroutines.jl [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+function get_basisidx(model::Model)
```
Comment (odow) [inline src/grb_advroutines.jl]: `c_api_getbasishead`

File: src/grb_advroutines.jl (hunk 2 above)
Comment (Thuener) [reply src/grb_advroutines.jl]: c_api_abinvrowi?

File: src/grb_advroutines.jl (hunk 3 above)
Comment (Thuener) [reply src/grb_advroutines.jl]: Do you prefer to remove get_tableaurow! ?

**Pull request closed without merging**